sklearn/utils/_random.pyx

```diff
@@ -314,6 +314,56 @@ cdef _sample_without_replacement(default_int n_population,
 
 def sample_without_replacement(
         object n_population, object n_samples, method="auto", random_state=None):
+    """Sample integers without replacement.
```

Comment:
This is the docstring of the private function above.

Reply:
Maybe we could remove the docstring from the private function and instead add an inline comment such as: "See the docstring of the public sample_without_replacement function." or similar.

Reply:
Done in my last commit

Reply:
I have to say it has been a while since I remember seeing such a confusing diff :sweat_smile: 

Looking at the file this is what we want, see https://github.com/glemaitre/scikit-learn/blob/a466d0489137fb0bd2c0d3fe5deefad2cf1d35a1/sklearn/utils/_random.pyx#L225 if you want to double-check

Reply:
Indeed it was a weird one. Thanks @lesteve to address the remaining comment.